code/src/components/Task.jsx

```diff
@@ -0,0 +1,261 @@
+import React, { useState } from "react";
+import { formatRelative } from "date-fns";
+import { useDispatch, useSelector } from "react-redux";
+import styled from "styled-components";
+import { keyframes } from "styled-components";
+import { projects } from "reducers/projects";
+import trashgray from "../assets/trashgray.svg";
+import checkblack from "../assets/checkblack.svg";
+import saveblack from "../assets/saveblack.svg";
+import editblack from "../assets/editblack.svg";
+
+const Task = ({ taskid, projectindex }) => {
+  const [editable, setEditable] = useState(false);
+  const [newText, setNewText] = useState("");
+  const dispatch = useDispatch();
+  const thisTask = useSelector((store) =>
+    store.projects.project[projectindex].tasks.find(
+      (task) => taskid === task.taskid
+    )
+  );
+  const taskIndex = useSelector((store) =>
+    store.projects.project[projectindex].tasks.findIndex(
+      (task) => taskid === task.taskid
+    )
+  );
+  const created = formatRelative(new Date(thisTask.added), new Date());
+
+  const toggleComplete = () => {
+    dispatch(
+      projects.actions.toggleTask({
+        taskindex: taskIndex,
+        projectindex: projectindex,
+      })
+    );
+  };
+
+  const deleteTask = () => {
+    dispatch(
+      projects.actions.deleteTask({
+        taskindex: taskIndex,
+        projectindex: projectindex,
+      })
+    );
+  };
+
+  const saveNewText = () => {
+    dispatch(
+      projects.actions.editTask({
+        taskindex: taskIndex,
+        projectindex: projectindex,
+        title: newText,
+      })
+    );
+  };
+
+  const editTask = () => {
+    setEditable(true);
+  };
+
+  const changingText = (e) => {
+    setNewText(e.currentTarget.textContent);
+  };
+
+  const pasteAsPlainText = (event) => {
+    event.preventDefault();
+    const text = event.clipboardData.getData("text/plain");
```

Comment:
Line 66 and 67 would be nice to comment what these do to your code. :)

Reply:
Ah yes, I see that. This is for when people copy-paste text into the to-do list, that it strips everything away and only returns plain text, instead of formatting, or other strange things that happen when you copy-paste directly into the DOM.